src/audio/eq_fir.c

```diff
@@ -515,7 +513,7 @@ static int fir_cmd_set_data(struct comp_dev *dev,
 	int ret = 0;
 
 	/* Check version from ABI header */
-	if (cdata->data->comp_abi != SOF_EQ_FIR_ABI_VERSION) {
+	if (cdata->data->comp_abi != SOF_ABI_VERSION) {
```

Comment:
The new ABI version has major, minor, micro versioning to allow addition of new IPCs without breaking backwards compatibility. The intention is that the ABI should not change frequently and we should not have sub ABIs

Reply:
I mean that changes can happen to EQs e.g. once in next few years time if some important need cannot be addressed with this EQ parameters notation. I added this check to avoid audio tuning accidents.

It's not clear to me how to be backwards compatible in EQs. Maybe there could be conversion function from old config to new algo version but it could become complex and 100% compatibility could need algo duplication. But it's doable if important.

I can change my review to accept.

But if we go this way, isn't this check in each component unnecessary? The component.h inline code could check this for all components.

Reply:
@singalsu we should have only a single ABI version for the whole FW otherise it becomes a burden tracking ABIs of each component. The major, minor, micro allows you to extend data members in the EQ structs (at the end) without breaking existing users. Same method is used by shared libraries.